lint-install.go

```diff
@@ -185,11 +185,12 @@ func goLintCmd(root string, level string) string {
 		suffix = " || true"
 	}
 
-	if len(found) == 1 && found[0] == root {
+	klog.Infof("found %d modules within %s: %s", len(found), root, found)
+	if len(found) == 0 || (len(found) == 1 && found[0] == strings.Trim(root, "/")) {
 		return fmt.Sprintf("out/linters/golangci-lint-$(GOLINT_VERSION)-$(LINT_ARCH) run%s", suffix)
 	}
 
-	return fmt.Sprintf(`find . -name go.mod | xargs -n1 dirname | xargs -n1 -I{} sh -c "cd {} && golangci-lint run -c $(GOLINT_CONFIG)"%s`, suffix)
+	return fmt.Sprintf(`find . -name go.mod | xargs -n1 dirname | xargs -n1 -I{} sh -c "cd {} && $(LINT_ROOT)/out/linters/golangci-lint-$(GOLINT_VERSION)-$(LINT_ARCH) run -c $(GOLINT_CONFIG)"%s`, suffix)
```

Comment:
This could also be
```
find . -name go.mod -execdir $(LINT_ROOT)/out/linters/golangci-lint-$(GOLINT_VERSION)-$(LINT_ARCH) run -c $(GOLINT_CONFIG)%s ;
```
and looks like:
```
[16:54:46]-[~/t/tt]─[manny@dellnix]> 
find
.
./a
./a/go
./a/go/mod1
./a/go/mod1/go.mod
./a/go/mod2
./a/go/mod2/another
./a/go/mod2/another/mod
./a/go/mod2/another/mod/go.mod
./a/go/mod2/another/go.mod
./a/go/mod2/go.mod
./a/go/go.mod
./a/go.mod
                                                                                                                                                                                      
[16:54:53]-[~/t/tt]─[manny@dellnix]> 
find -name go.mod -execdir touch golangci-lint-was-here ';'
                                                                                                                                                                                      
[16:54:56]-[~/t/tt]─[manny@dellnix]> 
find
.
./a
./a/go
./a/go/mod1
./a/go/mod1/go.mod
./a/go/mod1/golangci-lint-was-here
./a/go/mod2
./a/go/mod2/another
./a/go/mod2/another/mod
./a/go/mod2/another/mod/go.mod
./a/go/mod2/another/mod/golangci-lint-was-here
./a/go/mod2/another/go.mod
./a/go/mod2/another/golangci-lint-was-here
./a/go/mod2/go.mod
./a/go/mod2/golangci-lint-was-here
./a/go/go.mod
./a/go/golangci-lint-was-here
./a/go.mod
./a/golangci-lint-was-here
```

I think just find is easier to not mess up (shell quoting...), and also I'm currently getting the following annoying warning from xargs:
```
$ find . -name go.mod | xargs -n1 dirname | xargs -n1 -I{} sh -c "echo {}"
xargs: warning: options --max-args and --replace/-I/-i are mutually exclusive, ignoring previous --max-args value
./a/go/mod1
./a/go/mod2/another/mod
./a/go/mod2/another
./a/go/mod2
./a/go
./a
```

Reply:
TIL about execdir. Thank you! I've made the change.